pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -19,18 +19,27 @@ import (
 	"strings"
 
 	forwarder "github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	logsConfig "github.com/DataDog/datadog-agent/comp/logs/agent/config"
 	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/config/resolver"
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
 	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	logshttp "github.com/DataDog/datadog-agent/pkg/logs/client/http"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
 func init() {
 	diagnosis.Register("connectivity-datadog-core-endpoints", diagnose)
 }
 
+func getLogsHTTPEndpoints() (*logsConfig.Endpoints, error) {
+	// For now, a few parameters are hardcoded
+	datadogConfig := config.Datadog
+	logsConfigKey := logsConfig.NewLogsConfigKeys("diagnose_key", datadogConfig)
+	return logsConfig.BuildHTTPEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", "diagnose-track-type", logsConfig.DefaultIntakeProtocol, logsConfig.IntakeOrigin(logsConfig.DefaultIntakeProtocol))
```

Comment:
```suggestion
	return logsConfig.BuildHTTPEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", "logs", config.AgentJSONIntakeProtocol, config.DefaultIntakeOrigin)
```

We need to make sure the track type is still `logs` as this [changes the target URL](https://github.com/DataDog/datadog-agent/blob/781208614bbe4e61ee8e85985c968baaa8319795/pkg/logs/client/http/destination.go#L381). 

Also suggested updating the intake protocol and origin to match what the logs agent uses internally.

Reply:
fixed!